dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -1490,4 +1341,176 @@ Set getContainers() {
 	Set getFolders() {
 		return folders;
 	}
+
+	private class DependencyProcessorItem {
+		String assetKey;
+		AssetTypes assetTypes;
+
+		public DependencyProcessorItem(String assetKey, AssetTypes assetTypes) {
+			this.assetKey = assetKey;
+			this.assetTypes = assetTypes;
+		}
+	}
+
+	private class DependencyProcessor {
```

Comment:
Add Javadoc to the class and all of the methods in it

Reply:
done https://github.com/dotCMS/core/pull/19955/commits/2bba46387c6412fc00a7a021a2feae5e9d005d3c#diff-abb9df68dc16a9532f21f0164ea5c958a1f3462cce8aa09bcce2dd984da40910R1364